pkg/controller/controller.go

```diff
@@ -210,6 +225,48 @@ func createWorker(queue workqueue.RateLimitingInterface, resourceType string, ma
 	}()
 }
 
+func (c *controller) monitorConfigMap() {
+	// Cannot wait for the informer to push something into a queue.
+	// What we're waiting on may never exist without us configuring
+	// it, so we have to poll/ ask for it the first time to get it set.
+
+	// Can we ask 'through' an informer? Is it a writeback cache? I
+	// only ever want to monitor and be notified about one configmap
+	// in a hardcoded place.
+	glog.V(9).Info("cluster ID monitor loop enter")
+	cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
```

Comment:
I'm still not clear to me why we are polling every second for the cluster ID from the configmap. What are the deficiencies around having the controller fetch the cluster ID as needed instead of constantly? In other words, why can the controller not fetch (and set if needed) the cluster ID as part of the reconciling a service instance? Why does the cluster ID need to be stored by the controller object? Are we using that as some sort of backup storage for the cluster ID in case the configmap is deleted?

Reply:
Its kind of an interesting question. On the one hand if our controller is the only one who ever needs it then making it "call on use" would certainly work. However, there were some discussions around when someone may need to modify/set it - which would imply that there may be times to access it outside of the controller.

I'm thinking about start-up times... someone starts a new cluster and wants to set the cluser-id before any real work happens. In that case we'd need to make sure that the map and ID are there right away. That means run at least once at startup.  And what if someone deletes it? I kind of think we may want to recreate it quickly too for the same reasons - if someone goes looking for it but its not there that might be bad and confusing.

I'm less interested in logging that it changed but that's just me. If others see value in it then it can't hurt.

WDYT?

Reply:
does helm have access to a uuid generator? is that the appropriate place to try and make a fresh configmap? If it already exists, will it fail anything? I also don't want it to overwrite an existing one.